feat: add `stats/base/ndarray/drange`

gururaj1512 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Resolves None

## Description

> What is the purpose of this pull request?

This pull request:

-   feat: add `stats/base/ndarray/drange`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves None

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

gururaj1512 committed:
feat: add `stats/base/ndarray/drange`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/drange/README.md:
````diff
@@ -0,0 +1,115 @@
+```javascript
+var ndarray = require( '@stdlib/ndarray/base/ctor' );
+
+var xbuf = [ 1.0, 3.0, 4.0, 2.0 ];
````
@gururaj1512 This needs to be a `Float64Array`.

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/drange/docs/repl.txt:
```diff
@@ -0,0 +1,32 @@
+
+    Examples
+    --------
+    > var xbuf = [ 1.0, -2.0, 2.0 ];
```
@gururaj1512 This needs to be a `Float64Array`.

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/drange/docs/types/index.d.ts:
```diff
@@ -0,0 +1,45 @@
+
+/// <reference types="@stdlib/types"/>
+
+import { ndarray } from '@stdlib/types/ndarray';
```
```suggestion
import { float64ndarray } from '@stdlib/types/ndarray';
```

Should be updated below, as well.

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/drange/docs/types/index.d.ts:
```diff
@@ -0,0 +1,45 @@
+* @example
+* var ndarray = require( '@stdlib/ndarray/base/ctor' );
+*
+* var xbuf = [ 1.0, 3.0, 4.0, 2.0 ];
```
Incorrect.

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/drange/lib/index.js:
```diff
@@ -0,0 +1,44 @@
+* var ndarray = require( '@stdlib/ndarray/base/ctor' );
+* var drange = require( '@stdlib/stats/base/ndarray/drange' );
+*
+* var xbuf = [ 1.0, 3.0, 4.0, 2.0 ];
```
Incorrect.

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/drange/lib/main.js:
```diff
@@ -0,0 +1,55 @@
+* @example
+* var ndarray = require( '@stdlib/ndarray/base/ctor' );
+*
+* var xbuf = [ 1.0, 3.0, 4.0, 2.0 ];
```
Incorrect.

gururaj1512 committed:
chore: apply suggested changes

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (approved):
LGTM
